tapiro/core/src/main/scala/io/buildo/tapiro/Meta.scala

```diff
@@ -7,17 +7,19 @@ import scala.meta._
 import cats.data.NonEmptyList
 
 object Meta {
-  val codecsImplicits = (routes: List[TapiroRoute]) => routes.flatMap {
+  val codecsImplicits = (routes: List[TapiroRoute]) =>
+    routes.flatMap {
       case TapiroRoute(route, errorValues) =>
-        errorValues.map(_.name) ++
-          route.body.map(b => typeNameString(b.tpe)) :+
-          typeNameString(route.returns)
-    }.distinct.map(stringToImplicitParam)
-
-  private[this] val stringToImplicitParam = (name: String) => {
-    val paramName = Term.Name(s"${name.head.toLower}${name.tail}")
-    val nameType = Type.Name(name)
-    param"implicit ${paramName}: JsonCodec[$nameType]"
+        errorValues.map(m => MetarpheusType.Name(m.name)) ++
+          route.body.map(_.tpe) :+
+          route.returns
+    }.distinct.map(toImplicitParam)
+
+  private[this] val toImplicitParam = (`type`: MetarpheusType) => {
+    val typeName = typeNameString(`type`)
+    val paramName = Term.Name(s"${typeName.head.toLower}${typeName.tail}")
```

Comment:
I didn't change the parameter name here, so for `Id[Bot]` the parameter name is still `id` (i.e. the type names, first-letter-lowercased, disregarding the arguments).

With this approach we'll definitely run into collisions for routes that take both an `Id[A]` and an `Id[B]`, but I'm not sure what we should do about this.

We could concatenate the type name with the names of the arguments and get `idA`, `idB`, etc...

Alternatively, we could just generate dummy names (`x1`, ...) since I don't think we're actually using these names in the body, we're only using them implicitly...

Reply:
+1 for dummy names. I propose we open an issue and do that later, handling collisions it's not essential for success of the experiment.